### PR TITLE
feat: Set up Statistics page

### PR DESCRIPTION
**Pull request by bprusinowski**
This PR is a first step towards bringing more understanding about how Visualize evolves.

## How to test
**Statistics page**
1. Go to [this link](https://visualization-tool-git-feat-statistics-page-ixt1.vercel.app/statistics?dataSource=Prod).
2. See that the chart with basic statistics on the number of charts created on Vercel environments is shown (this page will show information for an environment it's deployed on, once we deploy to PROD, it will show PROD data).
3. Add a new chart using [this link](https://visualization-tool-git-feat-statistics-page-ixt1.vercel.app/?dataSource=Prod).
4. Refresh the Statistics page.
5. See that the number of charts increased 🎉
6. _Optional_: Add a `&flag__easter-eggs=true` to the end of the URL to see an alternative version of the chart.

### CODE REVIEW

File: app/pages/statistics.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  const now = Date.now();
+  const nowDate = new Date(now);
+  const lastMonthDailyAverage =
+    configsMetadata.filter(({ created_at: date }) => {
+      return date > new Date(now - 30 * DAYS_IN_MS) && date <= nowDate;
+    }).length / 30;
+  const previousThreeMonthsDailyAverage =
+    configsMetadata.filter(({ created_at: date }) => {
+      return date > new Date(now - 90 * DAYS_IN_MS) && date <= nowDate;
+    }).length / 90;
```
Comment (ptbrowne) [inline app/pages/statistics.tsx:46]: Nit: The computation of stats could be behind a big memo, and applied directly database side ?

Helped from ChatGPT gave me :

```
SELECT
    DATE_TRUNC('month', created_at) AS month,
    COUNT(*) AS monthly_count
FROM
    config
WHERE
    created_at >= DATE_TRUNC('month', CURRENT_DATE) - INTERVAL '1 year'
GROUP BY
    DATE_TRUNC('month', created_at)
ORDER BY
    month;
```

```
        month        | monthly_count 
---------------------+---------------
 2023-06-01 00:00:00 |             1
 2023-07-01 00:00:00 |             5
 2023-08-01 00:00:00 |             3
 2023-09-01 00:00:00 |             3
 2023-10-01 00:00:00 |             8
 2023-11-01 00:00:00 |             1
 2023-12-01 00:00:00 |             5
 2024-01-01 00:00:00 |             4
 2024-02-01 00:00:00 |            17
```

Comment (bprusinowski) [reply app/pages/statistics.tsx]: Optimized a bit in https://github.com/visualize-admin/visualization-tool/pull/1610/commits/5d48237f4f2d4998e5ab28888004ab194adfa12c 👍

File: app/pages/statistics.tsx [hunk 2]
```diff
@@ -116,6 +111,31 @@ const getCountByDate = (
...
+    data.filter(({ day }) => {
+      return day > new Date(now - 90 * DAYS_IN_MS) && day <= nowDate;
+    }).length / 90;
+  return {
+    direction:
+      lastMonthDailyAverage > previousThreeMonthsDailyAverage ? "up" : "down",
+    lastMonthDailyAverage,
+    previousThreeMonthsDailyAverage,
+  };
+};
```
Comment (ptbrowne) [inline app/pages/statistics.tsx:143]: For what's it's worth, the averages could also be directly be computed from the db

```
WITH
    last_month_daily_average AS (
        SELECT COUNT(*) / 30.0 AS daily_average
        FROM config
        WHERE
            created_at > CURRENT_DATE - INTERVAL '30 days'
            AND created_at <= CURRENT_DATE
    ),
    last_three_months_daily_average AS (
        SELECT COUNT(*) / 90.0 AS daily_average
        FROM config
        WHERE
            created_at > CURRENT_DATE - INTERVAL '90 days'
            AND created_at <= CURRENT_DATE
    )
SELECT
    (SELECT daily_average FROM last_month_daily_average) AS last_month_daily_average,
    (SELECT daily_average FROM last_three_months_daily_average) AS previous_three_months_daily_average;
```

**Review by ptbrowne — APPROVED** [review]
Thanks 👍 Awesome that we are starting to see this 💯

File: app/pages/statistics.tsx (hunk 2 above)
Comment (bprusinowski) [reply app/pages/statistics.tsx]: Thanks 💯 Done in https://github.com/visualize-admin/visualization-tool/pull/1610/commits/f234cb409c377857cf351fcb877de89040f7b29f 👍

**Pull request merged**